mesonbuild/dependencies/dev.py

```diff
@@ -530,6 +546,34 @@ def __init__(self, environment: 'Environment', kwargs: T.Dict[str, T.Any]):
         java_home_include = self.java_home / 'include'
         self.compile_args.append(f'-I{java_home_include}')
         self.compile_args.append(f'-I{java_home_include / platform_include_dir}')
+
+        if modules:
+            if m.is_windows():
+                java_home_lib = self.java_home / 'lib'
+                java_home_lib_server = java_home_lib
+            else:
+                if version_compare(self.version, '<= 1.8.0'):
+                    # The JDK and Meson have a disagreement here, so translate it
+                    # over. In the event more translation needs to be done, add to
+                    # following dict.
+                    def cpu_translate(cpu: str) -> str:
+                        java_cpus = {
+                            'x86_64': 'amd64',
+                        }
+
+                        return java_cpus.get(cpu, cpu)
+
+                    java_home_lib = self.java_home / 'jre' / 'lib' / cpu_translate(m.cpu_family)
+                    java_home_lib_server = java_home_lib / "server"
+                else:
+                    java_home_lib = self.java_home / 'lib'
+                    java_home_lib_server = java_home_lib / "server"
+
+            if 'jvm' in modules:
+                self.link_args.extend([f'-L{java_home_lib_server}', '-ljvm'])
+            if 'awt' in modules:
+                self.link_args.extend([f'-L{java_home_lib}', '-ljawt'])
```

Comment:
I just realized these should use `self.clib_compiler.find_library()`, not just because it "consistently" duplicates paths (by using filenames) but also because as we've seen, things might not necessarily be installed e.g. on Ubuntu package splitting.

So using find_library means we can actually do a configure check that these libraries exist, rather than failing at build time.